### PR TITLE
add support for new search filters and sort

### PR DESCRIPTION
**Pull request by toddmedema**
The API end of https://github.com/ExpeditionRPG/expedition-app/pull/341

### CODE REVIEW

File: models/quests.js [hunk 1]
```diff
@@ -66,18 +62,40 @@ exports.search = function(userId, params, callback) {
...
+      query = query.where('contentrating = ?', params.contentrating);
+    }
+
+    if (params.genre) {
+      query = query.where('genre = ?', params.genre);
+    }
+
     if (params.order) {
-      query = query.order(params.order.substr(1), (params.order[0] === '+'));
+      if (params.order === '+rating') {
```
Comment (smartin015) [inline models/quests.js]: `+contentrating`, to be consistent

File: models/quests.js [hunk 2]
```diff
@@ -66,18 +62,40 @@ exports.search = function(userId, params, callback) {
       query = query.where("published > NOW() - '? seconds'::INTERVAL", params.published_after);
     }
 
+    if (params.mintimeminutes) {
```
Comment (smartin015) [inline models/quests.js]: We _really_ need to develop a testable environment for SQL.

File: models/quests.js (hunk 1 above)
Comment (toddmedema) [inline models/quests.js]: Updated - actually `ratingavg` (contentrating is teen / everyone/ adult, this sorts by average review rating)

File: models/quests.js (hunk 2 above)
Comment (toddmedema) [inline models/quests.js]: Agreed. Added this to my list for the week. (also added stubs)

**Comment by 7thSentinel** [comment]
Gents, I'm still noticing an issue with the "App filter" when you search by "highest rating".  For instance, "Sally's Secret" (4 ratings, 1.5 star avg.) lists above "Bacchanalian Revels" (3 ratings, 3 star avg.), "(W.B. #2 Dark Gifts" (2 ratings, 4.5 star avg.) and "(Z.E. #2) Tensions Rising" (3 ratings, 4.5 star avg.). 

It's not because of # of ratings, because their are other quests with fewer ratings ranking above "Sally's Secret" (Spirit of Festivus) and it's not due to alphabetical listing...

**Pull request merged**